src/solvers/NelderMead.jl

```diff
@@ -80,62 +227,75 @@ function NelderMead!(
-    return o.x = o.population[argmin(o.costs)] # select min
+function initialize_solver!(mp::AbstractManoptProblem, s::NelderMeadState)
+    # init cost and p
+    s.costs = get_cost.(Ref(mp), s.population.pts)
+    return s.p = s.population.pts[argmin(s.costs)] # select min
 end
-function step_solver!(p::CostProblem, o::NelderMeadOptions, ::Any)
-    m = mean(p.M, o.population)
-    ind = sortperm(o.costs) # reordering for cost and p, i.e. minimizer is at ind[1]
-    ξ = inverse_retract(p.M, m, o.population[last(ind)], o.inverse_retraction_method)
+function step_solver!(mp::AbstractManoptProblem, s::NelderMeadState, ::Any)
+    M = get_manifold(mp)
+    m = mean(M, s.population.pts)
+    ind = sortperm(s.costs) # reordering for cost and p, i.e. minimizer is at ind[1]
+    ξ = inverse_retract(M, m, s.population.pts[last(ind)], s.inverse_retraction_method)
     # reflect last
-    xr = retract(p.M, m, -o.α * ξ, o.retraction_method)
-    Costr = get_cost(p, xr)
+    xr = retract(M, m, -s.α * ξ, s.retraction_method)
+    Costr = get_cost(mp, xr)
+    continue_steps = true
     # is it better than the worst but not better than the best?
-    if Costr >= o.costs[first(ind)] && Costr < o.costs[last(ind)]
+    if Costr >= s.costs[first(ind)] && Costr < s.costs[last(ind)]
         # store as last
-        o.population[last(ind)] = xr
-        o.costs[last(ind)] = Costr
+        s.population.pts[last(ind)] = xr
+        s.costs[last(ind)] = Costr
+        continue_steps = false
     end
     # --- Expansion ---
-    if Costr < o.costs[first(ind)] # reflected is better than fist -> expand
-        xe = retract(p.M, m, -o.γ * o.α * ξ, o.retraction_method)
-        Coste = get_cost(p, xe)
+    if Costr < s.costs[first(ind)] # reflected is better than fist -> expand
+        xe = retract(M, m, -s.γ * s.α * ξ, s.retraction_method)
+        Coste = get_cost(mp, xe)
         # successful? use the expanded, otherwise still use xr
-        o.population[last(ind)] .= Coste < Costr ? xe : xr
-        o.costs[last(ind)] = min(Coste, Costr)
+        s.population.pts[last(ind)] .= Coste < Costr ? xe : xr
+        s.costs[last(ind)] = min(Coste, Costr)
+        continue_steps = false
     end
     # --- Contraction ---
-    if Costr > o.costs[ind[end - 1]] # even worse than second worst
-        s = (Costr < o.costs[last(ind)] ? -o.ρ : o.ρ)
-        xc = retract(p.M, m, s * ξ, o.retraction_method)
-        Costc = get_cost(p, xc)
-        if Costc < o.costs[last(ind)] # better than last ? -> store
-            o.population[last(ind)] = xc
-            o.costs[last(ind)] = Costc
+    if continue_steps && Costr > s.costs[ind[end - 1]] # even worse than second worst
+        step = (Costr < s.costs[last(ind)] ? -s.ρ : s.ρ)
+        xc = retract(M, m, step * ξ, s.retraction_method)
+        Costc = get_cost(mp, xc)
+        if Costc < s.costs[last(ind)] # better than last ? -> store
+            s.population.pts[last(ind)] = xc
```

Comment:
Maybe `copyto!`would again be better / more general?

Reply:
I don't see how `copyto!` would be more general? This is just stores the entire point in an array.

Reply:
but FixedRankMatrices have points that are not arrays for example.

Reply:
I think that was resolved and discussed above.